events/iq_board.py

```diff
@@ -0,0 +1,94 @@
+import discord
+from discord.ext import commands
+import utils.logging as log
+import utils.colours as colour
+import utils.values as value
+import utils.database as sql
+from utils.helpers import *
+
+
+class IQBoard(commands.Cog):
+    def __init__(self, client):
+        self.client = client
+        self.iq_emoji = "<:WeSmart:682352802202386472>"
+        self.iq_requirement = 7
+
+    @commands.Cog.listener()
+    async def on_ready(self):
+        self.iq_channel = self.client.get_channel(-1)  # TODO: Create channel and change ID
+
+    @commands.Cog.listener()
+    async def on_reaction_add(self, reaction, user):
+        if str(reaction) == self.iq_emoji:
+            if reaction.count >= self.iq_requirement:
+                message = reaction.message
+                # Making sure people can't IQ Board IQ Boards
+                if message.channel == self.iq_channel:
+                    return
+                embed = self.create_iq_embed(message, reaction)
+                # Check if embed already exists in #iq-board
+                iq_embed = await self.find_iq_embed(embed)
+                if iq_embed is not None:
+                    await iq_embed.edit(embed=embed)
+                else:
+                    await self.iq_channel.send(embed=embed)
+
+    @commands.Cog.listener()
+    async def on_reaction_remove(self, reaction, user):
+        if str(reaction) == self.iq_emoji:
+            message = reaction.message
+            # Making sure people can't IQ Board IQ Boards
+            if message.channel == self.iq_channel:
+                return
+            new_embed = self.create_iq_embed(message, reaction)
+            old_embed = await self.find_iq_embed(new_embed)
+
+            # If it does not exist anymore we'll send it again
+            if old_embed is None and reaction.count > self.iq_requirement:
+                await self.iq_channel.send(embed=new_embed)
+            elif reaction.count == 0:
+                await old_embed.delete()
+            else:
+                await old_embed.edit(embed=new_embed)
+
+    @commands.Cog.listener()
+    async def on_message_delete(self, message):
+        reactions = [(str(r.emoji), r) for r in message.reactions]
+        if self.iq_emoji in [r[0] for r in reactions]:
+            embed_msg = await self.find_iq_embed(self.create_iq_embed(message, [r[1] for r in reactions if self.iq_emoji in r[0]][0]))
+            await embed_msg.delete()
+
+    # Find the already existing embed in the #iq-board channel
+    # arguments: discord.Embed() we just created
+    # return: None if the embed does not exist, if the embed/message exists it will return discord.Message
+    async def find_iq_embed(self, embed):
+        messages = [msg for msg in await self.iq_channel.history(limit=200).flatten() if msg.embeds]
```

Comment:
Could this become an issue if we end up with more than 200 IQ messages?

Reply:
I assume not many people would look at older IQ messages, maybe in a future update we could set the last embed as 'Outdated' or 'No longer updated'?